### PR TITLE
Move emit_hook to IpcServer

### PR DESCRIPTION
**Pull request by t-wissmann**
The hook communication is a core part of the ipc protocol and thus fits
perfectly into the IpcServer class. Note that on the client side, the
'hook' and 'call' are already grouped in a module ipc-server.c.

During this migration, some variable names changed:

  - g_event_window -> hookEventWindow_
  - last_property_number -> nextHookNumber_

### CODE REVIEW

**Comment by t-wissmann** [comment]
This includes #460

File: src/ipc-server.h [hunk 1]
```diff
@@ -2,22 +2,30 @@
...
     //! try to run an ipc request in the given window, return if there was one
     bool handleConnection(Window window);
+    //! send a hook to all listening clients
+    void emitHook(std::vector<std::string> args);
 
 private:
     XConnection& X;
+
+    Window hook_event_window_; //! window on which the hooks are announced
+    int next_hook_number_; //! index for the next hook
```
Comment (dnnr) [inline src/ipc-server.h]: Use camelCase here, as usual?

Comment (t-wissmann) [reply src/ipc-server.h]: Right! I've fixed this (also in this PR's description)

**Comment by t-wissmann** [comment]
I'm merging this since this is the approved PR modulo obvious merge conflict resolutions.

**Pull request merged**